x/group/spec/messages.md

```diff
@@ -0,0 +1,46 @@
+```go
+// Groups get their own GroupID
+type MsgCreateGroup struct {
+	Signer sdk.AccAddress `json:"signer"`
+	// The Owner of the group is allowed to change the group structure. A group account
+	// can own a group in order for the group to be able to manage its own members
+	Owner  sdk.AccAddress `json:"owner"`
+	// The members of the group and their associated weight
+	Members []Member `json:"members,omitempty"`
+	// TODO maybe make this something more specific to a domain name or a claim on identity? or Info leave it generic
+	Memo string `json:"memo,omitempty"`
+}
+
+// group accounts get their own sdk.AccAddress
+type MsgCreateGroupAccount struct {
+	Signer         sdk.AccAddress `json:"signer"`
+	// The Owner of a group account is allowed to change the DecisionPolicy. This can be left nil 
+	// in order for the group account to "own" itself
+	Owner          sdk.AccAddress `json:"owner"`
```

Comment:
Why is nil = group owns itself? Implicit nils are dangerous to me.
I would allow the options of no owner (can never change), self-owner, or external owner (maybe a 1 of N rule to move funds, controlled by a 3 of N rules to update the rules - or change the group).

At least in the state, nil should be no owner, rather than self-owner. Problem in the Msg is you don't know the self-address ahead of time. I would make that explicit somehow here (difference between self-owner and un-owned)

Reply:
And if there is an owner... we need a `MsgUpdateGroupAccount` to change the decision policy